src/firestore/encodeFirestoreValue.ts

```diff
@@ -0,0 +1,61 @@
+import * as _ from "lodash";
+
+import * as FirebaseError from "../error";
+
+function isPlainObject(input: any): boolean {
+  return (
+    typeof input === "object" &&
+    input !== null &&
```

Comment:
should the null check go before checking the type?

Reply:
this is the order it was before... `typeof null` actually returns `"object"` (fun fact) but I don't think the order of this matters much.